Cql/Cql.Compiler/ExpressionBuilderContext.cs

```diff
@@ -529,57 +547,87 @@ protected Expression Instance(Instance ine)
             }
         }
 
-        var parameterElements = ine.element!.SelectToArray(el => (el.name!, value:TranslateArg(el.value)));
+        (string name, Expression value)[] parameterNameValuePairs = ine.element!.SelectToArray(el => (name: el.name!, value: TranslateArg(el.value)));
 
-        var ctor = instanceType.GetConstructors()
-            .OrderByDescending(c => c.GetParameters().Length) // Prefer constructors with more parameters
-            .FirstOrDefault(c =>
-            {
-                var ctorParameters = c.GetParameters();
-
-                if (ctorParameters.Length < parameterElements.Length)
-                    return false; // Skip constructors with fewer parameters than provided
+        // Find a constructor that matches the provided parameters.
+        const int NOT_MAPPED_HAS_DEFAULT_VALUE = int.MaxValue;
+        const byte B_FALSE = 0x0, B_TRUE = 0x1;
+        var (ctor, ctorParameters, ctorPositionToParameterPositionMap) =
+            // Prefer constructors with more parameters, and
+            instanceType.GetConstructors()
+                        .OrderByDescending(c => c.GetParameters().Length) // Prefer constructors with more parameters
+                        .SelectWhere(ctor =>
+                        {
+                            var ctorParameters = ctor.GetParameters();
+
+                            // Exit if the constructor has fewer parameters than provided
+                            if (parameterNameValuePairs.Length > ctorParameters.Length)
+                                return default;
+
+                            // Exit if the constructor has a parameter that
+                            // is not assignable from the provided value by name or type
+                            // when the parameter has no default value.
+                            int[] ctorPositionToParameterPositionMap = new int[ctorParameters.Length];
+                            Span<byte> isParameterNameValuePairMapped = stackalloc byte[parameterNameValuePairs.Length];
```

Comment:
Is this stackalloc worth it? It looks like microoptimization to me, and results in code that others cannot maintain I am afraid.